src/components/structures/RoomDirectory.js

```diff
@@ -56,33 +56,10 @@ module.exports = React.createClass({
         });
     },
 
-    joinRoom: function(roomId, shouldPeek) {
-        var self = this;
-        self.setState({ loading: true });
-
-        var joinOrPeekPromise;
-
-        if (shouldPeek) {
-            joinOrPeekPromise = MatrixClientPeg.get().peekInRoom(roomId);
-        }
-        else {
-            joinOrPeekPromise = MatrixClientPeg.get().joinRoom(roomId);
-        }
-
-        joinOrPeekPromise.done(function() {
-            dis.dispatch({
-                action: 'view_room',
-                auto_peek: false, // don't peek as we've already peeked here (if it was needed)
-                room_id: roomId
-            });
-        }, function(err) {
-            console.error("Failed to join room: %s", JSON.stringify(err));
-            console.error(err);
-            var ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
-            Modal.createDialog(ErrorDialog, {
-                title: "Failed to join room",
-                description: err.message
-            });
+    showRoom: function(roomId) {
+        dis.dispatch({
+            action: 'view_room',
+            room_id: roomId
```

Comment:
At `:125` we pass a room alias into this function. Does this work? E.g. manually typing in a room alias into the box and hitting enter.